zio-http/src/main/scala/zio/http/Scheme.scala

```diff
@@ -41,41 +40,46 @@ sealed trait Scheme { self =>
     case _            => false
   }
 
+  /** defaultPort is only define for HTTP;HTTPS;WS;WSS */
   def defaultPort: Int = self match {
     case Scheme.HTTP  => 80
     case Scheme.HTTPS => 443
     case Scheme.WS    => 80
     case Scheme.WSS   => 443
+    case Scheme(_)    => 0
   }
 }
-object Scheme       {
+
+object Scheme {
 
   /**
    * Decodes a string to an Option of Scheme. Returns None in case of
    * null/non-valid Scheme
+   *
+   * Scheme = ALPHA *( ALPHA / DIGIT / "+" / "-" / "." )
```

Comment:
I was going to add a regex for this.
Also make sure everything is lowercase.

But I was concerned about efficient. Should we create a new method for that

Reply:
I am not sure if we should check the validity here. But I think we should ignore casing